iteration/iteration.go

```diff
@@ -26,19 +26,25 @@ const (
 	IterationStateClose    = "close"
 	PathSepInService       = "/"
 	PathSepInDatabase      = "."
+	IterationActive        = bool(true)
+	IterationNotActive     = bool(false)
 )
 
+var active bool
+
 // Iteration describes a single iteration
 type Iteration struct {
 	gormsupport.Lifecycle
-	ID          uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"` // This is the ID PK field
-	SpaceID     uuid.UUID `sql:"type:uuid"`
-	Path        path.Path
-	StartAt     *time.Time
-	EndAt       *time.Time
-	Name        string
-	Description *string
-	State       string // this tells if iteration is currently running or not
+	ID           uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"` // This is the ID PK field
+	SpaceID      uuid.UUID `sql:"type:uuid"`
+	Path         path.Path
+	StartAt      *time.Time
+	EndAt        *time.Time
+	Name         string
+	Description  *string
+	State        string // this tells if iteration is currently running or not
+	UserActive   *bool  // this can be nil
+	ActiveStatus bool
```

Comment:
I think scheduler will not be required for the gives use cases. In the current implementation, ActiveStatus is a write-only value, so that makes it redundant. So, let's remove ActiveStatus and always derive it based on other values.